File: services/dispute/src/main/java/com/example/iter/dispute/util/AdminReportMapper.java
package com.example.iter.dispute.util;

import com.example.iter.auth.api.UserSummary;
import com.example.iter.dispute.domain.entity.Report;
import com.example.iter.dispute.dto.response.AdminReportDetailResponse;
import com.example.iter.dispute.dto.response.ReportSummaryResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class AdminReportMapper {

    private final ReportMapper reportMapper;

    // 신고와 신고자 정보를 관리자 신고 목록 응답으로 변환합니다.
    public ReportSummaryResponse toSummary(Report report, UserSummary reporter) {
        return reportMapper.toSummary(report, reporter);
    }

    // 기본 신고 상세 정보에 관리자 처리 메모와 수정 시각을 추가합니다.
    public AdminReportDetailResponse toDetail(Report report, UserSummary reporter) {
        return new AdminReportDetailResponse(
                reportMapper.toDetail(report, reporter),
                report.getAdminMemo(),
                report.getUpdatedAt()
        );
    }
}

File: services/dispute/src/main/kotlin/com/example/iter/dispute/util/AdminReportMapper.kt
package com.example.iter.dispute.util

import com.example.iter.auth.api.UserSummary
import com.example.iter.dispute.domain.entity.Report
import com.example.iter.dispute.dto.response.AdminReportDetailResponse
import com.example.iter.dispute.dto.response.ReportSummaryResponse
import org.springframework.stereotype.Component

@Component
class AdminReportMapper(private val reportMapper: ReportMapper) {
    fun toSummary(report: Report, reporter: UserSummary): ReportSummaryResponse =
        reportMapper.toSummary(report, reporter)

    fun toDetail(report: Report, reporter: UserSummary): AdminReportDetailResponse = AdminReportDetailResponse(
        reportMapper.toDetail(report, reporter),
        report.adminMemo,
        report.updatedAt
    )
}
